services/wallet/controllers_v3.go

```diff
@@ -467,7 +467,8 @@ func LinkSolanaAddress(s *Service) handlers.AppHandler {
 
 		l := logging.Logger(ctx, "wallet")
 
-		if o := r.Header.Get("Origin"); o != s.dappConf.AllowedOrigin {
+		o := r.Header.Get("Origin")
+		if !isAllowedOrigin(o, s.dappConf.AllowedOrigin) {
```

Comment:
Nit: Since there could be multiple origins, should the config field be a plural?